packages/calcite-components/src/utils/dom.ts

```diff
@@ -661,27 +661,59 @@ export function isBefore(a: HTMLElement, b: HTMLElement): boolean {
  * @param animationName The name of the animation to watch for completion.
  */
 export async function whenAnimationDone(targetEl: HTMLElement, animationName: string): Promise<void> {
-  const { animationDuration: allDurations, animationName: allNames } = getComputedStyle(targetEl);
+  return whenTransitionOrAnimationDone(targetEl, animationName, "animation");
+}
+
+/**
+ * This util helps determine when a transition has completed.
+ *
+ * @param targetEl The element to watch for the transition to complete.
+ * @param transitionProp The name of the transition to watch for completion.
+ */
+export async function whenTransitionDone(targetEl: HTMLElement, transitionProp: string): Promise<void> {
```

Comment:
Can't we just use `onTransitionEnd` on a VNode in most cases? Should we specify that?

Reply:
I think we can tweak the name and doc to better explain its usage.